docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+  image.
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "unpaired" # one of "paired", "unpaired" or "grouped"
+  labeled: true
+  sample_label: "sample" # one of "sample", "all" or None
+  image_shape: [16, 16, 3]
+```
+
+##### Grouped
+
+- `intra_group_prob`: float, between 0 and 1. Passing 0 would only generate inter-group
+  samples, and passing 1 would only generate intra-group samples.
+- `sample_label`: method for sampling the labels "sample", "all"
+- `intra_group_option`: str, "forward", "backward, or "unconstrained"
+- `sample_image_in_group`: bool, if true, only one image pair will be yielded for each
+  group, so one epoch has num_groups pairs of data, if false, iterate through this
+  loader will generate all possible pairs.
+- `image_shape`: (list, tuple) len 3, corresponding to (dim1, dim2, dim3) of the 3D
+  image.
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "grouped" # one of "paired", "unpaired" or "grouped"
+  labeled: true
+  sample_label: "sample" # one of "sample", "all" or None
+  image_shape: [16, 16, 3]
+  sample_image_in_group: true
+  intra_group_prob: 0.7
+  intra_group_option: "forward"
+```
+
 See the [dataset loader configuration](dataset_loader.html) for more details.
 
 ## Train section
 
 The `train` section defines the neural network training hyper-parameters, by specifying
-subsections, `model`, `loss`, `optimizer`, `preprocess` and other training
-hyper-parameter, including `epochs` and `save_period`. See an
-[example configuration](https://github.com/DeepRegNet/DeepReg/blob/main/config/unpaired_labeled_ddf.yaml),
-with comments on the available options in each subsection.
+subsections, `method`, `backbone`, `loss`, `optimizer`, `preprocess` and other training
+hyper-parameters, including `epochs` and `save_period`.
+
+### Method - required
+
+The `method` argument defines the registration type. It must be a string type, one of
+"ddf", "dvf", "conditional", which are the currently supported registration methods.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+```
+
+### Backbone - required
+
+The `backbone` subsection is used to define the network, with all the network specific
+arguments under the same indent. The first argument should be the argument `name`, which
+should be string type, one of "unet", "local" or "global", to define a UNet, LocalNet or
+GlobalNet backbone, respectively. With Registry functionalities, you can also define
+your own networks to pass to DeepReg train via config.
```

Comment:
this needs a placeholder for link to registry

Reply:
the registry is not only for backbone @YipengHu, we should link on the top of this doc

https://deepreg.readthedocs.io/en/latest/docs/registry.html

btw, better to use relative link instead of URL link :)